### PR TITLE
test(elasticsearch): Add unit test for timestamp-based lineage feature

### PR DESCRIPTION
**Pull request by iprentic**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by RyanHolstien — APPROVED** [review]

File: metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java [hunk 1]
```diff
@@ -227,4 +230,142 @@ public void testConcurrentRemoveNodes() {
...
+        null,
+        null);
+    downstreamResult = getDownstreamLineage(datasetTwoUrn,
+        null,
+        null);
+    Assert.assertEquals(new Integer(1), upstreamResult.getTotal());
+    Assert.assertEquals(new Integer(2), downstreamResult.getTotal());
+
+    // Window includes initial time and updated time
+    upstreamResult = getUpstreamLineage(datasetTwoUrn,
```
Comment (jjoyce0510) [inline metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java:305]: Nice! Thanks for capturing all of these cases.

Are we able to add one where both createdAt and updatedAt timestamps are a) null or b) 0? We should treat either case as representing a "non-existant edge" (and disqualify it).

This is where we've had issues in the past! cc @aditya-radhakrishnan

Comment (iprentic) [reply metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java:306]: Yes! So just to clarify, these should not be returned for any searches no matter the timestamp, nor for searches with null timestamps?

**Comment by aditya-radhakrishnan** [comment]
I'm a bit confused, what's going on with the commit history here?

File: metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java (hunk 1 above)
Comment (aditya-radhakrishnan) [reply metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java:306]: Hmm actually @jjoyce0510 I don't think this is right. If both createdAt or updatedAt is 0 or null (falsey), we should return the edge.

Comment (jjoyce0510) [reply metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java:306]: Sorry - meant what aditya said. If BOTH have null or 0 value, we always want to include that edge in result set

Comment (iprentic) [reply metadata-io/src/test/java/com/linkedin/metadata/graph/elastic/ElasticSearchGraphServiceTest.java:306]: Great I will have a test for this!

**Comment by iprentic** [comment]
Updated here since I messed up this PR: https://github.com/datahub-project/datahub/pull/7661

**Pull request closed without merging**